pmd-java/src/main/java/net/sourceforge/pmd/lang/java/metrics/api/JavaClassMetricKey.java

```diff
@@ -101,7 +101,18 @@ public String name() {
             public boolean supports(ASTAnyTypeDeclaration node) {
                 return metric.supports(node);
             }
+
+
+            @Override
+            public boolean equals(Object obj) {
+                return obj == this;
```

Comment:
Is this correct? You could potentially create another key instance with the same name+metric. This method always returns a new instance, so this would fail:

    Metric metric = new CycloClassMetric();
    String name = "Cyclo";
    assert JavaClassMetricKey.of(metric, name)
        .equals(JavaClassMetricKey.of(metric, name));